File: chenxingyigou-parent/chenxingyigou-manager-web/src/main/webapp/js/controller/brandController.js
//创建控制器,$scope是控制层和视图层交流数据的桥梁
app.controller("brandController",function($scope,$http,brandService, $controller){

    $controller('baseController',{$scope:$scope});

    //查询品牌列表
    /*$scope.findAll=function () {
        brandService.findAll().success(function (response) {
             $scope.list=response;
         })
     }*/

    //分页
    /*  $scope.findPage=function(page,size){
          $http.get('../brand/findPage.do?page='+page+'&size='+size).success(
              function(response){
                  $scope.list=response.rows;//显示当前页的数据
                  $scope.paginationConf.totalItems=response.total;//更新总记录数
              }
          );
      }*/

    //新增，修改
    $scope.save=function () {
        var object=null;//方法名
        if ($scope.entity.id!=null){
            object=brandService.update($scope.entity);
        }else{
            object=brandService.add($scope.entity);
        }
        object.success(function (response) {
            if (response.success){
                alert(response.message);
                $scope.reloadList();//重新加载
            }else {
                alert(response.message);
            }
        })
    }
    //根据id查询品牌信息
    $scope.findOne=function (id) {
        brandService.findOne(id).success(function (response) {
            $scope.entity=response;
        })
    }

    //删除
    $scope.del=function () {
        brandService.del($scope.selectIds).success(function (response) {
            if (response.success){
                alert(response.message);
                $scope.reloadList();//重新加载
            } else {
                alert(response.message);
            }
        })
    }

    $scope.searchEntity={};
    //模糊查询
    $scope.search=function (page,size) {
        brandService.search(page,size,$scope.searchEntity).success(
            function(response){
                $scope.list=response.rows;//显示当前页的数据
                $scope.paginationConf.totalItems=response.total;//更新总记录数
            }
        );
    }
});

File: chenxingyigou-parent/chenxingyigou-manager-web/src/main/java/com/chenxingyigou/manager/controller/BrandController.java
package com.chenxingyigou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.chenxingyigou.pojo.TbBrand;
import com.chenxingyigou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    /**
     * 查询所有品牌信息
     * @return
     */
    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){

        return brandService.findAll();
    }

    /**
     * 分页查询品牌信息,被模糊查询替代
     * @param page
     * @param size
     * @return
     */
    @RequestMapping("/findPage")
    public PageResult findPage(int page,int size){
        return  brandService.findPage(page,size);
    }

    /**
     * 新增品牌信息
     * @param brand
     */
    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand brand){
        try {
            Boolean con = brandService.add(brand);
            if (con==true){
                return new Result(true,"保存成功！");
            }else {
                return new Result(false,"保存失败，该品牌已存在！");
            }
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"保存失败，服务器异常，请稍后再试！");
        }
    }

    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
      return  brandService.findOne(id);
    }

    @RequestMapping("/update")
    public  Result update(@RequestBody TbBrand brand){
        try {
            brandService.update(brand);
            return new Result(true,"修改成功！");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改失败，服务器异常，请稍后再试！");
        }
    }

    @RequestMapping("/delete")
    public Result delete(Long[] ids){
        try {
            brandService.delete(ids);
            return new Result(true,"删除成功！");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败，服务器异常，请稍后再试！");
        }
    }

    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand tbBrand,int page,int size){
       return brandService.findPage(tbBrand,page,size);
    }

    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList(){
        return brandService.selectOptionList();
    }
}
